render/src/umi/mod.rs

```diff
@@ -234,87 +332,144 @@ impl Umi {
                     if fetched.is_none() {
                         break;
                     }
-                    current_line = fetched.unwrap();
+                    current_line = fetched.unwrap_or_default();
                 }
 
-                Block::BulletListEntry(bullet_list_entry)
+                Ok(Block::BulletListEntry(bullet_list_entry))
             }
-            &_ => panic!(),
+            &_ => Err(UmiParserError::UnknownKind(current_line.position)),
         }
     }
 
-    pub fn create_um(&mut self) -> Document {
-        self.elements.clear();
-        debug_assert!(!self.ods.is_empty());
+    pub fn create_um(um_content: &str, config: &mut Config) -> Result<Document, UmiParserError> {
+        let mut umi = Umi::with_um(
+            vec![],
+            config.clone(),
+            config.preamble.i18n.lang.to_string().to_owned(),
+        );
+        umi.ods = um_content.into();
 
-        let wb: WorkBook = read_ods_buf(&self.ods).unwrap();
+        let wb: WorkBook = read_ods_buf(&umi.ods).unwrap_or_default();
         let sheet = wb.sheet(0);
         let rows = sheet.used_grid_size().0;
 
+        // Load Stored Config Values from Sheet
+        let hash_map_input: HashMap<String, String> = serde_json::from_str(
+            sheet
+                .cell(1, 3)
+                .unwrap_or_default()
+                .value
+                .as_str_opt()
+                .unwrap_or_default(),
+        )
+        .unwrap_or_default();
+        let input_path: PathBuf =
+            PathBuf::from(hash_map_input.get("input_path").unwrap_or(&String::new()));
+
+        umi.config.preamble = serde_yaml::from_str(
+            sheet
+                .cell(1, 4)
+                .unwrap_or_default()
+                .value
+                .as_cow_str_or("")
+                .to_string()
+                .as_str(),
+        )
+        .unwrap_or_default();
+        umi.config.input = input_path;
+
+        // Determine the correct column for parsing Locale Content and Attributes
+        let mut index = 5;
```

Comment:
```suggestion
        let mut index = DEFAULT_CONTENT_COLUMN;
```

use the newly defined const here too to better see the connection.